deploy/buildtools/buildconfig.py

```diff
@@ -1,177 +1,151 @@
+        s3_bucketname: S3 bucketname for AFI builds.
+        post_build_hook: Post build hook script.
+        build_farm_host: Name of build farm host type.
+        build_farm_host_dispatcher: Build farm host dispatcher object.
+    """
+    name: str
+    build_config_file: BuildConfigFile
+    TARGET_PROJECT: Optional[str]
+    DESIGN: str
+    TARGET_CONFIG: str
+    deploytriplet: Optional[str]
+    launch_time: str
+    PLATFORM_CONFIG: str
+    s3_bucketname: str
+    post_build_hook: str
+    build_farm_host: str
+    build_farm_host_dispatcher: BuildFarmHostDispatcher
+
+    def __init__(self, name: str, recipe_config_dict: Dict[str, Any], build_farm_hosts_config_file: Dict[str, Any], build_config_file: BuildConfigFile, launch_time: str) -> None:
+        """
+        Args:
+            name: Name of config i.e. name of `config_build_recipe.yaml` section.
+            recipe_config_dict: `config_build_recipe.yaml` options associated with name.
+            build_farm_hosts_config_file: Parsed representation of `config_build_farm_hosts.yaml` file.
+            build_config_file: Global build config file.
+            launch_time: Time manager was launched.
+        """
         self.name = name
-        self.TARGET_PROJECT = buildconfigdict.get('TARGET_PROJECT')
-        self.DESIGN = buildconfigdict['DESIGN']
-        self.TARGET_CONFIG = buildconfigdict['TARGET_CONFIG']
-        self.PLATFORM_CONFIG = buildconfigdict['PLATFORM_CONFIG']
-        self.instancetype = buildconfigdict['instancetype']
-        self.deploytriplet = buildconfigdict['deploytriplet']
+        self.build_config_file = build_config_file
+
+        self.TARGET_PROJECT = recipe_config_dict.get('TARGET_PROJECT')
+        self.DESIGN = recipe_config_dict['DESIGN']
+        self.TARGET_CONFIG = recipe_config_dict['TARGET_CONFIG']
+        self.deploytriplet = recipe_config_dict['deploy-triplet']
         self.launch_time = launch_time
-        self.launched_instance_object = None
 
-    def __repr__(self):
-        return "BuildConfig obj:\n" + pprint.pformat(vars(self), indent=10)
+        # run platform specific options
+        self.PLATFORM_CONFIG = recipe_config_dict['PLATFORM_CONFIG']
+        self.s3_bucketname = recipe_config_dict['s3-bucket-name']
+        if valid_aws_configure_creds():
+            aws_resource_names_dict = aws_resource_names()
+            if aws_resource_names_dict['s3bucketname'] is not None:
+                # in tutorial mode, special s3 bucket name
+                self.s3_bucketname = aws_resource_names_dict['s3bucketname']
+        self.post_build_hook = recipe_config_dict['post-build-hook']
+
+        # retrieve the build host section
+        self.build_farm_host = recipe_config_dict.get('build-farm', "default-build-farm")
+        build_farm_host_conf_dict = build_farm_hosts_config_file[self.build_farm_host]
+
+        build_farm_host_type = build_farm_host_conf_dict["build-farm-type"]
```

Comment:
I'm confused by the docstring for the `build_farm_host` attribute:
```
       build_farm_host: Name of build farm host type.
```
And the name of this local `build_farm_host_type`.  I think the names and descriptions here are a little too close together.  I need to reread the sample_config_*.yaml files again and come back here with a concrete suggestion.

Reply:
Cleared this up a bit in the code:

`build_farm_host_name` is the name of the build farm given by the user (i.e. "default-build-farm" or "local-build-farm")
`build_farm_host_type_name` is the human readable "NAME" attribute that links to a specific dispatcher class.